.vitepress/config.js

```diff
@@ -0,0 +1,172 @@
+                    [ 'meta', { name: 'citation_technical_report_institution', content: siteData.title } ],
+                    [ 'meta', { name: 'citation_technical_report_number', content: frontmatter.eip } ],
+                    // LD+JSON
+                    [ 'script', { type: 'application/ld+json' }, JSON.stringify({
+                        "@type": "WebSite",
+                        "url": `https://eips.ethereum.org/${pageData.relativePath}`,
+                        "name": eipTitle,
+                        "description": pageData.description,
+                        "@context": "https://schema.org"
+                    })],
+                ].filter(x => x?.length);
+            } else {
+                return [];
+            }
+        } catch (error) {
+            logger.error(error);
+            throw error;
+        }
+    },
+    async transformPageData(pageData) {
+        try { // Custom error handling needed because of the way VitePress handles runtime errors (i.e. it doesn't)
+            logger.info(`Transforming ${pageData.relativePath}`);
+            
+            pageData = { ...pageData };
+            let { frontmatter } = pageData;
+
+            if (frontmatter.eip) {
+                // Try to read from cache
+                try {
+                    let cache = JSON.parse(fs.readFileSync(`./.vitepress/cache/eips/${frontmatter.eip}.json`));
+                    frontmatter = { ...frontmatter, ...cache };
+                } catch (e) {
+                    logger.info(`Cache miss for ${pageData.relativePath}`);
+                }
+                // The below caused so much pain and suffering :|
+                if (!frontmatter.created) {
+                    let initial = new Date((await git.log(["--diff-filter=A", "--", pageData.relativePath])).latest.date); // Only one match, so this is fine to use latest
+                    if (initial) {
+                        frontmatter.created = initial.toISOString().split('T')[0];
+                    }
+                }
+                if (!frontmatter.finalized && frontmatter.status === 'Final') {
+                    let final = new Date((await git.raw(['blame', pageData.relativePath])).split('\n').filter(line => line.match(/status:\s+final/gi))?.pop()?.match(/(?<=\s)\d+-\d+-\d+/g)?.pop());
+                    if (final) {
+                        frontmatter.finalized = final.toISOString().split('T')[0];
+                    }
+                }
+                if (frontmatter.created.toISOString) { // It's a date object. We don't want that.
+                    frontmatter.created = frontmatter.created.toISOString().split('T')[0];
+                }
+
+                // Write to cache
+                if (!fs.existsSync('./.vitepress/cache/eips')) {
+                    fs.mkdirSync('./.vitepress/cache/eips', { recursive: true });
+                }
```

Comment:
We can avoid a TOCTOU here:

> [Calling `fsPromises.mkdir()` when `path` is a directory that exists results in a rejection only when `recursive` is false.](https://nodejs.org/api/fs.html#fsmkdirpath-options-callback)

Also, use the asynchronous version if you can.

```suggestion
                await fs.mkdir('./.vitepress/cache/eips', { recursive: true });
```

Reply:
Done.